GH#1737 Retain '@' when replacing usernames

cosimo opened the pull request:
# Description

See issue #1737 for context.

Currently it's not possible from the skill code to distinguish between `@botname message` and `botname message`. With this change, username mentions retain the `@` character.

Fixes #1737 



## Status
**READY** 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

It has not been tested.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

jacobtomlinson reviewed (approved):
I'm so sorry this took so long to review, things have been very quiet here recently. I've updated to the latest commit and fixed up a test. This should be good to merge.

cosimo commented:
> I'm so sorry this took so long to review, things have been very quiet here recently. I've updated to the latest commit and fixed up a test. This should be good to merge.

Thanks @jacobtomlinson for looking into this. For me this PR for part of preparation work to make our opsdroid skills deal in a more automatic way with '@' mentions.

With this PR merged it should be possible to use `@match_regex("@botname ...")`. However, I tried to understand whether the slack connector natively supported mentions, using the slack application `app_mention` events, https://api.slack.com/events/app_mention. From what I could understand, the opsdroid slack connector doesn't have such support yet.

Did you have any intentions to support mentions via app_mention events for the slack connector? Would you accept any contribution in that sense if I worked on it? 

(Assuming it's at all possible. So far I haven't seen any indication that slack would send app_mention events, I only received plain messages so far, no matter what I try).

jacobtomlinson commented:
> Would you accept any contribution in that sense if I worked on it?

Absolutely. I recommend you start by opening an issue to describe that functionality. Then I'd be more than happy to review a PR that adds it.
